client/log_verifier.go

```diff
@@ -133,7 +133,8 @@ func (c *logVerifier) BuildLeaf(data []byte) (*trillian.LogLeaf, error) {
 		return nil, err
 	}
 	return &trillian.LogLeaf{
-		LeafValue:      data,
-		MerkleLeafHash: leafHash,
+		LeafValue:        data,
+		MerkleLeafHash:   leafHash,
+		LeafIdentityHash: leafHash,
```

Comment:
This isn't necessarily the case; it's only equivalent to `MerkleLeafHash` iff the personality hasn't defined it to be otherwise.

Reply:
Would `sha256` of data be better?
Maybe in the future we can add hooks into the client that will allow personalities to customize it. At the moment though, that doesn't seem to be needed. 

The `LeafIdentityHash` is required to be present in prod. This change is to just have _something_ in this field.